src/parse.rs

```diff
@@ -0,0 +1,407 @@
+use crate::{
+    errors::{Error, Result},
+    RSAPrivateKey, RSAPublicKey,
+};
+use simple_asn1::{ASN1Block, ASN1DecodeErr, BigUint, OID};
+
+use std::convert::TryFrom;
+
+impl From<ASN1DecodeErr> for Error {
+    fn from(e: ASN1DecodeErr) -> Error {
+        Error::ParseError {
+            reason: format!("{}", e),
+        }
+    }
+}
+
+#[cfg(feature = "pem")]
+impl TryFrom<pem::Pem> for RSAPrivateKey {
+    type Error = Error;
+
+    /// Parses a `PKCS8` or `PKCS1` encoded RSA Private Key.
+    ///
+    /// Expects one of the following `pem` headers:
+    /// - `-----BEGIN PRIVATE KEY-----`
+    /// - `-----BEGIN RSA PRIVATE KEY-----`
+    fn try_from(pem: pem::Pem) -> Result<RSAPrivateKey> {
```

Comment:
could you add a doctest for this, so it is easy to see in the docs how it works?